src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -169,6 +185,61 @@ LogEventPropertyValue CreatePropertyValue(object? value, Destructuring destructu
         return new ScalarValue(value.ToString() ?? "");
     }
 
+    Destructuring? GetCachedFallbackDestructuring(
+#if !NET5_0
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+#endif
+        Type type)
+    {
+        if (_fallbackDestructuringCache.TryGetValue(type, out var fallbackDestructuring)) return fallbackDestructuring;
+
+        var fallback = GetFallbackDestructuring(type);
+        _fallbackDestructuringCache[type] = fallback;
```

Comment:
This is only done for performance to not resolve the whole type hierarchy on every conversion. Overwriting it even on multi-threading would not lead to race conditions as the value would also be the same.